CONTRIBUTING.md

```diff
@@ -0,0 +1,61 @@
+
+2. Get your local development build working by following our [README](README.md). Fork our repo and make sure to merge our staging branch into yours **WEEKLY** as we do update frequently.
+
+3. [Join our community development HipChat room](http://www.hipchat.com/gBebkHP6g) for help getting set up. We're happy to have you! If you want to make sure you get our attention, include an **@all** (everyone) or **@here** (everyone currently in the room) in your message.
+
+## Code Style
+
+Please make sure your code conforms to our [style guide](STYLEGUIDE.md).  We have implemented linting to help you with this:
+
+* To install our precommit hook that will lint the files you've changed, run `rake install:hooks`. You should only have to do this once.  We _highly_ recommend it!
+* To lint the entire project, run `rake lint` from the repository root.  This will check all Ruby, JavaScript and HAML.
+* To lint only part of the project, run one of these commands from the repository root:
+  * dashboard: See the [dashboard README](./dashboard#code-style)
+  * pegasus Ruby: `cd pegasus && rubocop`
+  * pegasus HAML: `haml-lint pegasus`
+  * apps: See the [apps README](./apps#style-guide)
+  * code-studio: See the [code-studio README](./code-studio#code-style)
+
+## Testing
+
+Please test your changes before submitting them to us!
+
+### Manually
+
+We support recent versions of Firefox, Chrome, IE9, iOS Safari and the Android browsers ([full list of supported browsers and versions](https://support.code.org/hc/en-us/articles/202591743)). Be sure to try your feature out in [IE9](docs/testing-ie9.md), iOS and Android if it's a risk. [BrowserStack live](http://www.browserstack.com) or [Sauce Labs manual](https://saucelabs.com/manual) let you run manual tests in these browsers remotely.
+
+### Unit tests
+
+For dashboard changes, be sure to test your changes using `rake test`. For [apps or blockly](./apps) changes, see our [grunt testing instructions](./apps#running-tests).
+
+### UI tests
+
+Our continuous integration server regularly runs a suite of [UI tests](./dashboard/test/ui) using Selenium / Cucumber which run against many browsers via [BrowserStack Automate](https://www.browserstack.com/automate), and can also be run locally using `chromedriver`. See the [README](./dashboard/test/ui) in that folder for instructions.
+
+If your changes might affect level paths, blockly UI, or critical path site logic, be sure to test your changes with a local UI test.
+
+## Submitting your Pull Request
+
+Once you've linted and tested your changes, send us a pull request!  Contributors should follow the GitHub [fork-and-pull model](https://help.github.com/articles/using-pull-requests) to submit pull requests into this repository.
```

Comment:
I'll add a note to that effect; you mean the comment to be directed to internal contributors, correct? Because external contributors (using fork-and-pull) don't have permission to push to staging.